src/test/java/com/gavinfenton/quizolation/service/TeamServiceTest.java

```diff
@@ -0,0 +1,115 @@
+package com.gavinfenton.quizolation.service;
+
+import com.gavinfenton.quizolation.entity.Team;
+import com.gavinfenton.quizolation.repository.TeamRepository;
+import org.hibernate.ObjectNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class TeamServiceTest {
+
+    @InjectMocks
+    private TeamService teamService;
+
+    @Mock
+    private TeamRepository teamRepository;
+
+    @BeforeEach
+    public void setup() {
+        initMocks(this);
+    }
+
+    @Test
+    public void testCreateTeam() {
+        Team teamSaving = new Team();
+        teamSaving.setName("Team name");
+        Team teamExpected = new Team();
+
+        given(teamRepository.save(teamSaving)).willReturn(teamExpected);
+
+        Team teamActual = teamService.createTeam(teamSaving);
+
+        verify(teamRepository).save(teamSaving);
+        assertEquals(teamExpected, teamActual);
+    }
+
+    @Test
+    public void testUpdateTeam() {
+        Team teamSaving = new Team();
+        teamSaving.setId(1L);
+        teamSaving.setName("Team name");
+        Long idExpected = teamSaving.getId();
+        Team teamExpected = new Team();
+
+        given(teamRepository.save(teamSaving)).willReturn(teamExpected);
+
+        Team teamActual = teamService.updateTeam(idExpected, teamSaving);
+        assertEquals(teamActual, teamExpected);
+    }
+
+    @Test
+    public void testGetTeam() {
+        Team teamSaving = new Team();
+        teamSaving.setId(1L);
+        teamSaving.setName("Team name");
```

Comment:
Not saving anything so this entity isn't needed

Reply:
You copy & pasted the teams test for this, so getTeam() isn't being tested any more